src/common/BaseCommand.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { SfCommand } from '@salesforce/sf-plugins-core';
+import { Logger } from '@salesforce/core';
+import { CommandLineUtils } from './Common';
+import { LoggerSetup } from './LoggerSetup';
+import { HasRequirements, CommandRequirements } from './Requirements';
+
+export abstract class BaseCommand
+    extends SfCommand<any>
+    implements HasRequirements
+{
+    protected _commandName = 'BaseCommand';
+    public get commandName(): string {
+        return this._commandName;
+    }
+
+    protected _flagValues: any;
+    public get flagValues(): any {
+        return this._flagValues;
+    }
+
+    protected _logger!: Logger;
+    public get logger(): Logger {
+        return this._logger;
+    }
+
+    protected _commandRequirements: CommandRequirements = {};
+    public get commandRequirements(): CommandRequirements {
+        return this._commandRequirements;
+    }
+
+    protected populateCommandRequirements(): void {
+        // override in child classes and update _commandRequirements
+        // to include whatever requirements the command has.
+    }
+
+    public async init(): Promise<void> {
+        if (this.logger) {
+            // already initialized
+            return Promise.resolve();
+        }
+
+        this.injectFlagParser();
+
+        return super
+            .init()
+            .then(() => this.parse())
+            .then((parserOutput) => {
+                this._flagValues = parserOutput.flags;
+                return Logger.child(this.commandName);
+            })
+            .then((logger) => {
+                this._logger = logger;
+                return LoggerSetup.initializePluginLoggers();
+            })
+            .then(() => this.populateCommandRequirements());
+    }
+
+    // Loops over all of the flags of a command and checks to see
+    // if a flag has a `validate` function defined. If so then we
+    // mark it to use `CommandLineUtils.flagParser` as a parser,
+    // which will execute the validation function.
+    private injectFlagParser() {
+        const flags = (<typeof SfCommand>this.constructor).flags;
+        const flagEntries = Object.entries(flags);
+        flagEntries.forEach((flag) => {
+            const configEntries = Object.entries(flag[1]);
```

Comment:
What is `flag[1]`? The value associated with the flag?

Reply:
`Object.entries` returns an array of KeyValue pairs where item[0] is key and item[1] is value. In this context `flag[1]` holds another object which defines the configs for that flag. By default, `oclif` flag configs contain a parser that just accepts whatever value is passed in for that flag. Here in this function we check to see if `validate` is added to the configs for a flag. If so then we override the default parser with our parser which runs the validation check and only accepts the flag value if validation passes.

Reply:
Do you think it would be helpful if we put the above in comment?

Reply:
You mean in addition to the comment that I've already put on the method? I guess I can reword that comment to include more info.

Reply:
It would be a nice addition, I think, although a nit, yes. I don't anticipate working with flags is going to be the dominant work in lwc-dev-mobile-core in the future but if anyone needs to visit and find what's going on right away a little note should be beneficial.